javalib/src/main/scala/java/util/AbstractMap.scala

```diff
@@ -0,0 +1,199 @@
+package java.util
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+
+object AbstractMap {
+
+  private def entryEquals[K, V](entry: Map.Entry[K, V], other: Any): Boolean = {
+    other match {
+      case other: Map.Entry[_, _] =>
+        entry.getKey == other.getKey && entry.getValue == other.getValue
```

Comment:
You sure can implement it in a companion object, or even in the package object, but make it `private[util]`.

Note that you can use an `implicit class` here, which was precisely invented for this use case.

I'm just a bit wary of the particular choice of `===`. This operator means something in JS, and it's not the same, which could be troubling. That said, I don't really have a better suggestion. My least worst idea is `java_==`, but it's not very nice. So let's keep yours unless you have a better idea.